### DISCUSSION TITLE
Next Release of Fairlearn (v0.5.1 vs 0.6.0)

### DISCUSSION THREAD

**Discussion by riedgar-ms** [fairlearn/fairlearn#688] [General]
As mentioned in the community call, I think that we should do another release of Fairlearn. Back in #659, we were talking about doing monthly releases, and we're now coming up on three since the last one. I have a PR (#686 ) out to update the Changes doc, since some things got missed.

I don't think there are any major changes which could be merged imminently, so I don't think there's a particular reason for further delay. The one potential one is the change of `master` to `main`. However, I'd suggest that we do this release, then rename the branch, officially check in the governance docs, and release 0.6.0 then.

Thoughts? @hildeweerts @adrinjalali @romanlutz @MiroDudik @mmadaio

**Comment by romanlutz** [comment]
I'm in favor of 0.6.0 since we have a new module (preprocessing).

I agree on waiting with `master` -> `main` until this is done to avoid interrupting the release pipeline. As soon as you do the release I can take care of the `master` -> `main` switch.

What is the second release in your sequence?

**Reply by riedgar-ms** [reply]
The second release would be the official "Fairlearn is now separate from Microsoft" release. It might not do much more than that (apart from the branch name switch).

**Reply by romanlutz** [reply]
Why does there need to be a release without real new content just because of a governance change? It's not like the license changes. I may very well be missing something significant, so please point it out to me. If anyone feels strongly about this I'm not going to insist either. It's just that as a user I expect there to be changes, especially when the minor version changes and not just patch.

**Reply by riedgar-ms** [reply]
A release just for 'Fairlearn is now independent' is certainly not necessary. I thought it might be nice to put in a clear demarcation, but you have a good point about nothing changing in the package or the licence. Consider that part very much a suggestion to be discussed :-)

**Reply by romanlutz** [reply]
It's absolutely worth a discussion! Thanks for raising this! Also, maybe my view with Microsoft glasses on is very different from others' views. Perhaps this is a decision that I/we should recuse my/ourselves from and let @hildeweerts @adrinjalali make the choice without my/our bias. :-)

**Reply by hildeweerts** [reply]
I actually don't really have an opinion on this (very off-brand, I know). So I'm happy either way :)

**Reply by MiroDudik** [reply]
I think we should do release now -- I would lean towards 0.5.1, but I don't care too much.

There are no clean-cut semantic versioning rules for 0.y.z, but we have been approximately following this:
* breaking changes: increment y
* non-breaking changes: suffices to increment z

So the next obvious breaking change is going to be when the widget goes away.

When we update the governance and put it in .RST, I think we should bump up z, just so that the website is accurately reflecting the current governance. [Edited: I just noticed the suggestion to recuse ourselves... I'll stay recused!]

**Reply by adrinjalali** [reply]
I'd say if we're adding at least one new feature, it should be 0.6 and not 0.5.1, since the latter implies mere fixes IMO.

As for the governance change, I think we should just include it in whatever release comes after the change, w/o taking the governance change into account in the release schedule.

**Reply by riedgar-ms** [reply]
Given the various votes and non-votes, I think the consensus is for v0.6.0 (since the new release will have the correlation remover) and no extra release for the governance change?

**Reply by MiroDudik** [reply]
Sounds good to me.

**Reply by adrinjalali** [reply]
I'd say the change in the governance is effective whenever we merge PRs related to them. From the users' perspective, governance doesn't change anything in the library, and therefor shouldn't change the version number alone.
